projects/jsf-validation/src/lib/schema-helper.service.ts

```diff
@@ -37,7 +39,20 @@ export class SchemaHelperService {
         let isEmpty = true;
         for (const p of Object.keys(cur)) {
           isEmpty = false;
-          recursiveFlattening(cur[ p ], prop ? `${prop}.${p}` : p);
+          const childProp = prop ? `${prop}.${p}` : p;
+          recursiveFlattening(cur[ p ], childProp);
+
+          // Enums are always required
+          if (cur[p][SchemaHelperService.ENUM_KEY] && !SchemaHelperService.hasKey(flattenedObject, childProp, cur, p, SchemaHelperService.READONLY_KEY)) {
+            const existingRequiredKeys = SchemaHelperService.getExistingKeys(flattenedObject,  prop, SchemaHelperService.REQUIRED_KEY);
+            if (!existingRequiredKeys.length) {
+              SchemaHelperService.setArrayValue(flattenedObject, prop, SchemaHelperService.REQUIRED_KEY, p);
+            } else {
+              if (existingRequiredKeys.every(key => flattenedObject[key] !== p)) {
+                SchemaHelperService.setArrayValue(flattenedObject, prop, SchemaHelperService.REQUIRED_KEY, p);
+              }
+            }
```

Comment:
Can we simplify it to this?
```suggestion
            if (!existingRequiredKeys.length || existingRequiredKeys.every(key => flattenedObject[key] !== p)) {
              SchemaHelperService.setArrayValue(flattenedObject, prop, SchemaHelperService.REQUIRED_KEY, p);
            } 
```